Source/SVWidgetsLib/FilterParameterWidgets/ImportHDF5DatasetWidget.cpp

```diff
@@ -0,0 +1,675 @@
+
+  m_Filter = dynamic_cast<ImportHDF5Dataset*>(filter);
+  Q_ASSERT_X(m_Filter != nullptr, "NULL Pointer", "ImportHDF5DatasetWidget can ONLY be used with an ImportHDF5Dataset filter");
+
+  setupUi(this);
+  setupGui();
+
+  if(filter)
+  {
+    QString currentPath = filter->property(PROPERTY_NAME_AS_CHAR).toString();
+    if(currentPath.isEmpty() == false)
+    {
+      currentPath = QDir::toNativeSeparators(currentPath);
+      // Store the last used directory into the private instance variable
+      QFileInfo fi(currentPath);
+      m_OpenDialogLastDirectory = fi.path();
+    }
+    else
+    {
+      m_OpenDialogLastDirectory = QDir::homePath();
+    }
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+ImportHDF5DatasetWidget::~ImportHDF5DatasetWidget()
+{
+  if(m_FileId > 0)
+  {
+    H5Fclose(m_FileId);
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void ImportHDF5DatasetWidget::setupGui()
+{
+  // Catch when the filter is about to execute the preflight
+  connect(getFilter(), SIGNAL(preflightAboutToExecute()), this, SLOT(beforePreflight()));
+
+  // Catch when the filter is finished running the preflight
+  connect(getFilter(), SIGNAL(preflightExecuted()), this, SLOT(afterPreflight()));
+
+  // Catch when the filter wants its values updated
+  connect(getFilter(), SIGNAL(updateFilterParameters(AbstractFilter*)), this, SLOT(filterNeedsInputParameters(AbstractFilter*)));
+
+  connect(dsetPathLE, &QLineEdit::textChanged, [=] { emit parametersChanged(); });
+
+  // When a dataset in the treeview is double-clicked, write its path to the dataset path line edit
+  connect(hdfTreeView, &QTreeView::doubleClicked, [=](const QModelIndex& index) {
+    ImportHDF5TreeModel* treeModel = dynamic_cast<ImportHDF5TreeModel*>(hdfTreeView->model());
+    if(treeModel != nullptr)
+    {
+      if(treeModel->hasChildren(index) == false)
+      {
+        dsetPathLE->setText(treeModel->hdfPathForIndex(index));
+      }
+    }
+  });
+
+  // Allow the widget to fill the entire FilterInputWidget space
+  setWidgetIsExpanding(true);
+
+  QFont inputFileFont;
+  inputFileFont.setBold(true);
+  inputFileFont.setItalic(true);
+  inputFileFont.setWeight(75);
+  inputFileFont.setStyleStrategy(QFont::PreferAntialias);
+#if defined(Q_OS_MAC)
+  inputFileFont.setPointSize(12);
+#elif defined(Q_OS_WIN)
+  inputFileFont.setPointSize(9);
+#else
+  inputFileFont.setPointSize(10);
+#endif
+
+  value->setFont(inputFileFont);
+  value->hide();
+
+  if(m_Filter != nullptr)
+  {
+    QString dsetPath = m_Filter->getDatasetPath();
+    dsetPathLE->setText(dsetPath);
+
+    QString hdf5FilePath = m_Filter->getHDF5FilePath();
+    if(hdf5FilePath.isEmpty() == false)
+    {
+      value->setText(hdf5FilePath);
```

Comment:
What happens if the file does not exist? There needs to be a check to ensure there is a file at the location in the hdf5FilePath variable.

Just drop the filter into a pipeline. it will crash.